openmc/lib/mesh.py

```diff
@@ -123,6 +142,67 @@ def id(self):
     def id(self, mesh_id):
         _dll.openmc_mesh_set_id(self._index, mesh_id)
 
+    @property
+    def n_elements(self):
+        n = c_size_t()
+        _dll.openmc_mesh_get_n_elements(self._index, n)
+        return n.value
+
+    def material_volumes(
+            self,
+            n_samples: int = 10_000,
+            prn_seed: Optional[int] = None
+    ) -> List[List[Tuple[Material, float]]]:
+        """Determine volume of materials in each mesh element
+
+        .. versionadded:: 0.14.1
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples in each mesh element
+        prn_seed : int
+            Pseudorandom number generator (PRNG) seed; if None, one will be
+            generated randomly.
+
+        Returns
+        -------
+        List of tuple of (material, volume) for each mesh element. Void volume
+        is represented by having a value of None in the first element of a
+        tuple.
+
+        """
+        if n_samples <= 0:
+            raise ValueError("Number of samples must be positive")
```

Comment:
should we perhaps add another check that checks the number of samples is above the number of mesh voxels and perhaps returns a warning

Reply:
This is the number of samples _per_ mesh element, so such a warning wouldn't make sense.

Reply:
In that case should this number always be greater than 1?

Reply:
It's up to the user to decide how many samples they think are necessary. If you had a mesh that was exactly conformal to material boundaries and had exactly one material per mesh element, using a single sample in each mesh element would yield the correct volumes. Of course, that's a pretty contrived example but just pointing out that really the only constraint (which is enforced here) is that the number of samples needs to be positive. Obviously for practical cases you would want to use a lot more than 1 sample per element to get accurate values.